### DISCUSSION TITLE
Github Desktop | Cannot add [bob] to .gitignore

### DISCUSSION THREAD

**Discussion by realtkco** [community/community#22199] [New to GitHub]
<p>Hi, First time posting on here.</p>
<p>I’m looking on help with github desktop, Just trying to create a .gitignore on a file “resource/[bob]/” and it seems to not be working but for “db/” it does work, Any suggestions? I can’t change the name of the folder btw.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/3/93ae4f884e4e0b686b19a89231ea3381901d390d.png" data-download-href="/uploads/short-url/l4rE7gzzyaB0P50ewDyyF5PMoqh.png?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/37914184/181069457-0aa139ca-ce47-496d-b20d-647225ff7cf6.png" alt="" data-base62-sha1="l4rE7gzzyaB0P50ewDyyF5PMoqh" width="341" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/3/93ae4f884e4e0b686b19a89231ea3381901d390d_2_341x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/3/93ae4f884e4e0b686b19a89231ea3381901d390d_2_511x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/3/93ae4f884e4e0b686b19a89231ea3381901d390d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/3/93ae4f884e4e0b686b19a89231ea3381901d390d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">517×757 40.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks!</p>

**Comment by airtower-luna** [comment]
<p>Anything enclosed in square brackets will be interpreted as range notation, as described in the <a href="https://git-scm.com/docs/gitignore#_pattern_format">gitignore pattern format</a> documentation. So <code>[bob]</code> will match a <code>b</code> or <code>o</code> character. If you want to match literal square brackets you need to escape them with backslashes:</p>
<pre><code class="lang-nohighlight">resource/\[bob\]/
</code></pre>

**Comment by realtkco** [comment]
<p>Seems to not work, I was looking into pattern format but I have not got that far.</p>
<p>That did not work what should I do now?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Hm, I wonder if that has anything to do with you working on Windows, where the backslash can also be a directory separator? Not sure.</p>
<p>Maybe try using this:</p>
<pre><code class="lang-nohighlight">resource/?bob?/
</code></pre>
<p>The <code>?</code> matches any character other than <code>/</code>, if you don’t have any other directories that match (e.g. <code>resource/abobc/</code>) it should work.</p>

**Comment by realtkco** [comment]
<p>I have found out with that info, That did not work but just messing around with it <code>?bob</code> works!</p>
<p>Thanks!</p>
